includes/class-wc-cart-session.php

```diff
@@ -38,7 +38,12 @@ public function __construct( &$cart ) {
 		}
 
 		$this->cart = $cart;
+	}
 
+	/**
+	 * Register methods for this object on the appropriate WordPress hooks.
+	 */
+	public function register_hooks() {
```

Comment:
What if instead of adding the burden of calling this manually everywhere cart/session is used, we add a second arg to the constructor. If true (default), register hooks. If false, do not.

Then the clone method can simply pass false to the 2nd arg.

Reply:
Generally, it's much better to avoid registering hooks within a constructor. There are [numerous](https://carlalexander.ca/designing-class-wordpress-hooks/) [resources](https://tommcfarlin.com/wordpress-plugin-constructors-hooks/) [available](https://twitter.com/schlessera/status/902827389939539973) that explain why this is the case. Especially since this is in the context of unit testing, I feel like it's easy to see why an explicit method call is preferable.

With that said, I respect that I don't own this project, and if your preference is a constructor argument instead of a separate method, I'll make that change.

Reply:
With that in mind then, shall we use `init` instead of `register_hooks` - other WC classes use that naming. `register_hooks` is something unique to this PR.